test/integration/truncate_test.lua

```diff
@@ -0,0 +1,124 @@
+local fio = require('fio')
+
+local t = require('luatest')
+local g_memtx = t.group('truncate_memtx')
+local g_vinyl = t.group('truncate_vinyl')
+
+local helpers = require('test.helper')
+
+math.randomseed(os.time())
+
+local function before_all(g, engine)
+    g.cluster = helpers.Cluster:new({
+        datadir = fio.tempdir(),
+        server_command = helpers.entrypoint('srv_select'),
+        use_vshard = true,
+        replicasets = {
+            {
+                uuid = helpers.uuid('a'),
+                alias = 'router',
+                roles = { 'customers-router' },
+                servers = {
+                    { instance_uuid = helpers.uuid('a', 1), alias = 'router' },
+                },
+            },
+            {
+                uuid = helpers.uuid('b'),
+                alias = 's-1',
+                roles = { 'customers-storage' },
+                servers = {
+                    { instance_uuid = helpers.uuid('b', 1), alias = 's1-master' },
+                    { instance_uuid = helpers.uuid('b', 2), alias = 's1-replica' },
+                },
+            },
+            {
+                uuid = helpers.uuid('c'),
+                alias = 's-2',
+                roles = { 'customers-storage' },
+                servers = {
+                    { instance_uuid = helpers.uuid('c', 1), alias = 's2-master' },
+                    { instance_uuid = helpers.uuid('c', 2), alias = 's2-replica' },
+                },
+            }
+        },
+        env = {
+            ['ENGINE'] = engine,
+        },
+    })
+    g.engine = engine
+    g.cluster:start()
+
+    g.space_format = g.cluster.servers[2].net_box.space.customers:format()
+end
+
+g_memtx.before_all = function() before_all(g_memtx, 'memtx') end
+g_vinyl.before_all = function() before_all(g_vinyl, 'vinyl') end
+
+local function after_all(g)
+    g.cluster:stop()
+    fio.rmtree(g.cluster.datadir)
+end
+
+g_memtx.after_all = function() after_all(g_memtx) end
+g_vinyl.after_all = function() after_all(g_vinyl) end
+
+local function before_each(g)
+    for _, server in ipairs(g.cluster.servers) do
+        server.net_box:eval([[
+            local space = box.space.customers
+            if space ~= nil and not box.cfg.read_only then
+                space:truncate()
+            end
+        ]])
+    end
+end
+
+g_memtx.before_each(function() before_each(g_memtx) end)
+g_vinyl.before_each(function() before_each(g_vinyl) end)
+
+local function add(name, fn)
```

Comment:
Sometimes this function should become a helper...

Reply:
I think we've discussed it before